
# SILVI
A  pipeline to facilitate in T-cell epitope selection 

## Citation
The paper describing this pipeline is available [here](link)

## Installation

Clone the repository locally with the command:
```
git clone https://github.com/loire/SILVI.git
```

## Purpose

Merge and filter results of immunoinformatics tools to facilitate discovery of epitopes

## Workflow

![workflow](images/workflow.png)

### R package dependancies:

tidyverse
stringr

### General execution process

The script Fire.R contains the commands to integrate initial data and apply analysis steps. 
Path to input files should be edited by the user in Fire.R 
We recommand loading the script Fire.R in Rstudio (link to rstudio) and execute each step sequentially. 

## Step1

### Input

FOR CLASS I:

First, users must provide results for proteins of interest generated by epitope binding programs.
Submit fasta sequences of proteins to the following online predictors
Currently, the script only accept results from 9-mer epitopes for class I alleles.

links to online predictor:
[]()
[]()

Format your results files as csv 
* 4 columns: allele,peptides,seq_num,score
* comma separated
* Files name **MUST** be formated as: proteincode_predictorcode.csv 
* Move your files in a directory named data created in the working directory

FOR CLASS II: 


Similarly, user should obtained results from the online predictors.
Currently, the script only accept results from 15-mer  epitopes for Class II alleles.

links to online predictor:
[]()
[]()

Format your results files as csv 
* 4 columns: allele,seq_num,full_peptide,rank,core,ic50
* semi-colon separated
* Files name **MUST** be formated as: proteincode_predictorcode.csv 
* Move your files in a directory named data created in the working directory



### Intermediate Output

For  class I, the first step of the script output the following files:
* 1_common_I.csv
	Formated prediction for class I supertype in a csv file.
* 1_blast_me.fasta
	Fasta file of peptitdes sequence to blast again host proteome.

For  class II, the first step of the script output the following files:
* 1_common_II.csv
	Formated prediction for class II supertype in a csv file.
* 1_blast_me.fasta
	Fasta file of peptides sequence to blast again host proteome.



